src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -779,19 +778,33 @@ bool OutputStateMachineEngine::ActionOscDispatch(const wchar_t /*wch*/,
             TermTelemetry::Instance().Log(TermTelemetry::Codes::OSCWT);
             break;
         case OscActionCodes::SetColor:
-            success = _dispatch->SetColorTableEntry(tableIndex, color);
+            for (size_t i = 0; i < tableIndexes.size(); i++)
+            {
+                auto tableIndex = tableIndexes.at(i);
+                auto rgb = colors.at(i);

```

Comment:
because you've validated the lengths of these arrays, you can use `til::at(tableIndexes, i)` and `til::at(colors, i)`